cpp/zotaut/sync_legacy_records.cc

```diff
@@ -0,0 +1,279 @@
+/** \brief Utility for syncing legacy records from the FID stock to zotero delivered records
+ *  \author Johannes Riedl (johannes.riedl@uni-tuebingen.de)
+ *
+ *  \copyright 2024 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <algorithm>
+#include <iostream>
+#include <map>
+#include <unordered_set>
+#include <cstdio>
+#include <cstdlib>
+#include <nlohmann/json.hpp>
+#include "Compiler.h"
+#include "DbConnection.h"
+#include "DbResultSet.h"
+#include "DbRow.h"
+#include "DnsUtil.h"
+#include "EmailSender.h"
+#include "IniFile.h"
+#include "MARC.h"
+#include "RegexMatcher.h"
+#include "Solr.h"
+#include "SolrJSON.h"
+#include "StringUtil.h"
+#include "UBTools.h"
+#include "ZoteroHarvesterUtil.h"
+#include "util.h"
+
+// Needed because of a clang bug (c.f. https://github.com/llvm/llvm-project/issues/42943) that
+// prevents properly detecting that LegacyEntryInformationHash and LegacyEntryInformationEqual below are needed
+#pragma clang diagnostic ignored "-Wunneeded-internal-declaration"
+
+namespace {
+
+
+[[noreturn]] void Usage() {
+    ::Usage("[--zotero-conf path] [--harvester-conf path] (--all (Beware - long runtime!!) | journal_name)");
+}
+
+
+struct LegacyRecordStock {
+    std::string ixtheo_host_;
+    std::string ixtheo_port_;
+    std::string krim_host_;
+    std::string krim_port_;
+};
+
+
+struct LegacyEntryInformation {
+    std::vector<std::string> dois_;
+    std::string main_title_;
+    std::string record_id_;
+};
+
+
+auto LegacyEntryInformationHash([](const LegacyEntryInformation &A) { return std::hash<std::string>()(StringUtil::Join(A.dois_, "")); });
+auto LegacyEntryInformationEqual([](const LegacyEntryInformation &A, const LegacyEntryInformation &B) { return A.dois_ == B.dois_; });
+using LegacyEntryInformationSet =
+    std::unordered_set<LegacyEntryInformation, decltype(LegacyEntryInformationHash), decltype(LegacyEntryInformationEqual)>;
+
+
+const std::string CONF_FILE_PATH(UBTools::GetTuelibPath() + "zotero.conf");
+const std::string LEGACY_RECORD_SECTION_NAME("LegacyRecordStock");
```

Comment:
Shouldn't the logic for parsing the legacy section be moved to `ZoteroHarvesterConfig.h/cc`? Not sure whether processing the config with this section might fail in other programs if the section is unknown to the regular config parser